### PR TITLE
Recipient can view list of available foodbags

### PR DESCRIPTION
**Pull request by swardbladpaulo**
[API: Recipient can view a list of available FoodBags](https://www.pivotaltracker.com/story/show/176614254)
### User Story
```
As a recipient,
In order to choose the perfect bag for me
I want to be able to see a list of available bags
```
## Changes proposed in this pull request:
* ams.rb, add gem active_model_serializers, add index action to foodbags_controler and modify before_action, created foodbags_index_serializer, modify routes, modify users.rb email to be randomnized, create spec visitor_can_view_index_list_of_foodbags correct routes with create add dateformatted to serializer
* correct routes with create
* add dateformatted to serializer
## What have we learned working on this feature:
* How to display the BE to Mobile
* To add serializers
## Packages/Gems added
* active_model_serializers

### CODE REVIEW

File: spec/factories/users.rb [hunk 1]
```diff
@@ -1,6 +1,6 @@
 FactoryBot.define do
   factory :user do
-    email { 'user_becomes@donor.com' }
+    email { "user_becomes#{rand(0..999)}@donor.com" }
```
Comment (Ehugo2000) [inline spec/factories/users.rb:3]: smart!

**Review by Ehugo2000 — COMMENTED** [review]
nice!

File: spec/requests/api/visitor_can_view_index_list_of_foodbags_spec.rb [hunk 2]
```diff
@@ -0,0 +1,17 @@
+RSpec.describe "GET /api/foodbags" do
+  let!(:foodbags) {5.times{create(:foodbag)}}
```
Comment (Ehugo2000) [inline spec/requests/api/visitor_can_view_index_list_of_foodbags_spec.rb:2]: maybe a test for list of several foodbags..

File: app/serializers/foodbags_index_serializer_serializer.rb [hunk 3]
```diff
@@ -0,0 +1,8 @@
+class FoodbagsIndexSerializerSerializer < ActiveModel::Serializer
+    attributes :id, :pickuptime, :status
+    attribute :created_at do
+    object.created_at.to_formatted_s(:db)
```
Comment (Ehugo2000) [inline app/serializers/foodbags_index_serializer_serializer.rb:4]: what is this? just to understand :)

**Review by Ehugo2000 — COMMENTED** [review]
nice!

File: config/routes.rb [hunk 4]
```diff
@@ -1,7 +1,7 @@
 Rails.application.routes.draw do
-  mount_devise_token_auth_for 'User', at: 'api/auth'
+  mount_devise_token_auth_for 'User', at: 'api/auth', skip: [:omniauth_callbacks]
  
   namespace :api do
-    resources :foodbags, only: [:create]
+    resources :foodbags, only: [:index, :show, :create]
```
Comment (Ehugo2000) [inline config/routes.rb:5]: show! now? :)

File: app/serializers/foodbags_index_serializer_serializer.rb (hunk 3 above)
Comment (swardbladpaulo) [reply app/serializers/foodbags_index_serializer_serializer.rb:4]: This is to format the date to yyyy-mm-dd hh:mm:ss

File: config/routes.rb (hunk 4 above)
Comment (swardbladpaulo) [reply config/routes.rb:5]: We need to show the foodbags on the FE

File: spec/requests/api/visitor_can_view_index_list_of_foodbags_spec.rb (hunk 2 above)
Comment (swardbladpaulo) [reply spec/requests/api/visitor_can_view_index_list_of_foodbags_spec.rb:2]: Why do you think that? My thoughts is that is if its fly once, it should fly 5 times.

**Pull request merged**